pkg/apis/serving/v1beta1/explainer_custom.go

```diff
@@ -69,6 +75,22 @@ func (c *CustomExplainer) GetStorageSpec() *StorageSpec {
 // GetContainer transforms the resource into a container spec
 func (c *CustomExplainer) GetContainer(metadata metav1.ObjectMeta, extensions *ComponentExtensionSpec, config *InferenceServicesConfig) *v1.Container {
 	container := &c.Containers[0]
+
+	argumentPredictorHost := constants.PredictorURL(metadata, false)
+	deploymentMode, ok := metadata.Annotations[constants.DeploymentMode]
+
+	if ok && (deploymentMode == string(constants.ModelMeshDeployment)) {
```

Comment:
same here to call [GetDeploymentMode](https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/utils/utils.go#L68)

Reply:
GetContainer is part of serving component interface, not acting as a controller/reconciler. The serving components do not have the controller client or DeployConfig which is required by the global GetDeploymentMode function. I am not sure whether it is necessary to use the utility function, which seems designed for controllers, in serving components.